src/plugins/saved_objects/public/finder/saved_object_finder.tsx

```diff
@@ -268,259 +223,159 @@ class SavedObjectFinderUi extends React.Component<
     );
   };
 
-  private getAvailableSavedObjectMetaData() {
-    const typesInItems = new Set<string>();
-    this.state.items.forEach((item) => {
-      typesInItems.add(item.type);
-    });
-    return this.props.savedObjectMetaData.filter((metaData) => typesInItems.has(metaData.type));
-  }
+  private fetchAllowedTypes = async () => {
+    const allowedTypes = await getSavedObjects().getAllowedTypes();
+    this.setState({ allowedTypes });
+  };
 
-  private getSortOptions() {
-    const sortOptions = [
-      <EuiContextMenuItem
-        key="asc"
-        icon={
-          this.state.sortDirection === 'asc' ||
-          (this.state.query === '' && this.state.sortDirection !== 'desc')
-            ? 'check'
-            : 'empty'
+  // server-side paging not supported
+  // 1) saved object client does not support sorting by title because title is only mapped as analyzed
+  // 2) can not search on anything other than title because all other fields are stored in opaque JSON strings,
+  //    for example, visualizations need to be search by isLab but this is not possible in Elasticsearch side
+  //    with the current mappings
+  public render() {
+    const { onChoose, savedObjectMetaData } = this.props;
+    const taggingApi = getSavedObjects().getSavedObjectsTagging();
+    const originalTagColumn = taggingApi?.ui.getTableColumnDefinition();
```

Comment:
I think it would be valuable (and less risky) to add the feature as an explicit opt-in, by adding a new boolean prop/flag to the finder's props, wdyt?

Reply:
Since I created a new plugin and finder instead of modifying the existing one, this didn't seem necessary anymore so I left it out.